spec/jobs/digital_object_import_processing_job_spec.rb

```diff
@@ -2,7 +2,185 @@
 
 require 'rails_helper'
 
-RSpec.describe DigitalObjectImportProcessingJob do
-  context '.process' do
+RSpec.describe DigitalObjectImportProcessingJob, solr: true do
+  let(:user) { FactoryBot.create(:user) }
+  let(:batch_import) { FactoryBot.create(:batch_import, user: user) }
+  let(:digital_object_import) { FactoryBot.create(:digital_object_import, batch_import: batch_import) }
+  let(:digital_object_import_id) { digital_object_import.id }
+
+  let(:second_digital_object_import) { FactoryBot.create(:digital_object_import, batch_import: batch_import, index: 2) }
+  let(:third_digital_object_import) { FactoryBot.create(:digital_object_import, batch_import: batch_import, index: 3) }
+
+  let(:created_digital_object) { DigitalObject::Base.find(DigitalObjectRecord.first.uid) }
+
+  context '.perform' do
+    before do
+      allow(described_class).to receive(:find_digital_object_import).and_return(digital_object_import)
+      allow(described_class).to receive(:queue_applicable_import_prerequisites).and_call_original
+      allow(described_class).to receive(:apply_recursive_failure!).and_call_original
+      allow(digital_object_import).to receive(:in_progress!).and_call_original
+      described_class.perform(digital_object_import_id)
+    end
+
+    context 'successful run' do
+      it 'marks the DigitalObjectImport as in_progress during processing' do
+        expect(digital_object_import).to have_received(:in_progress!)
+      end
+
+      it 'results in a DigitalObjectImport status of success at the end of the job' do
+        expect(digital_object_import.status).to eq('success')
+      end
+
+      it 'results in a DigitalObjectImport with no import_errors' do
+        expect(digital_object_import.import_errors).to be_blank
+      end
+
+      context 'creates the expected digital object' do
+        it "is of the expected type" do
+          expect(created_digital_object).to be_a(DigitalObject::Item)
+        end
+
+        it "has the expected dyanmic field data" do
+          expect(created_digital_object.dynamic_field_data).to eq(
+            'title' => [{ 'sort_portion' => 'The', 'non_sort_portion' => 'Cool Item' }],
+            'abstract' => [{ 'abstract_value' => 'some abstract' }]
+          )
+        end
+
+        it "is marked as created and updated by the expected user" do
+          expect(created_digital_object.created_by).to eq(nil)
```

Comment:
Is the created_by/updated_by user not set? I am probably missing something, sorry.

Reply:
I'm glad you pointed this out!  That test was definitely not testing the right thing, and revealed a bug.  Just fixed it with my latest commit.  Thanks!